### PR TITLE
Support migrations using non-disk storage #1094

### PR DESCRIPTION
**Pull request by timbru**
Note: we need to release kvx first.

### CODE REVIEW

File: src/commons/util/mod.rs [hunk 1]
```diff
@@ -36,6 +35,11 @@ impl KrillVersion {
         Self::from_str(KRILL_VERSION).unwrap()
     }
 
+    /// Make a notation friendly to namespaces for upgrades.
+    pub fn hyphen_notated(&self) -> String {
+        format!("{}-{}-{}{}", self.major, self.minor, self.patch, self.release_type)
```
Comment (Koenvh1) [inline src/commons/util/mod.rs:40]: There's in principle nothing that guarantees that `self.release_type` will be in a format friendly to namespaces, but I guess we are not going to ever put something silly in there, and if we do we will probably notice it anyway.

**Review by Koenvh1 — APPROVED** [review]
Looks good to me. There are two very very very minor things that you should feel free to ignore.

File: src/commons/util/mod.rs (hunk 1 above)
Comment (timbru) [reply src/commons/util/mod.rs:40]: In principle, these values come from the `version` entry in Cargo.toml. And we use a sane scheme for the release_type postfix.

File: src/upgrades/mod.rs [hunk 2]
```diff
@@ -918,82 +815,71 @@ pub fn finalise_data_migration(
...
+                    if let Some(signer_handle) = &openssl_signer_handle {
+                        let internal_key_id = key_id.to_string();
+                        if let Some(mapper) = krill_signer.get_mapper() {
+                            mapper.add_key(signer_handle, &key_id, &internal_key_id)?;
+                            num_recorded_keys += 1;
+                        }
+                    }
+                }
+            } else {
+                debug!("Could not parse key as key identifier: {}", key);
```
[the remark this answers is not in the document]
Comment (timbru) [reply src/upgrades/mod.rs:874]: Yeah, that's okay - it means that someone mucked about with files on disk manually. So, they are kind of on their own then anyway.

**Pull request merged**